### PR TITLE
add explicit prompts to templates for product impact & tracking issues

### PR DESCRIPTION
**Pull request by haszari**
This PR adjusts the templates for issues and PRs to add prompts to explictly consider what products are impacted.

The goal is to help get in the habit of considering the following, as part of any change:

- Which products need the fix or feature (e.g. WC Pay, WC Subscriptions, or both).
- How to reproduce the issue in each affected product.
- How to verify the fix in each affected product.
- Who has the ball for releasing the fix, when it will go out to customers (etc).

Ideally we'd consider this whenever we log an issue, and especially important when we start work on a PR.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -17,32 +18,26 @@ Questions for PR author:
...
-
-<!--
-Make sure you edit the page for the current release when adding testing instructions.
-We often create a blank page ahead of time for the next release.
--->
+## Product impact
+<!-- What products will this PR ship in? -->
 
+- [ ] Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref
+- [ ] Will this PR affect WooCommerce Payments? yes/no/tbc, add issue ref
```
Comment (brucealdridge) [inline .github/PULL_REQUEST_TEMPLATE.md:44]:
```suggestion
- [ ] Added changelog entry (or does not apply)
- [ ] Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref
- [ ] Will this PR affect WooCommerce Payments? yes/no/tbc, add issue ref
```

**Review by brucealdridge — APPROVED** [review]
LGTM, have suggested adding in changelog checklist.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (haszari) [reply .github/PULL_REQUEST_TEMPLATE.md:44]: Good thinking!

Though I figure the changelog entries will get added to the product repos directly. We need to nail down the changelog process.

**Comment by haszari** [comment]
Re-requested reviews since this has been sitting a while and it's process-related - more eyes is a good thing. Feel free to merge on my behalf :)

**Comment by james-allan** [comment]
This looks good to me. I saw the suggestion of adding a changelog section however given we decided not to include the changelog in the PR description, I don't think that is relevant anymore. 

> Feel free to merge on my behalf :)

Okey dokey.

**Pull request merged**